exercises/forth/cases_test.go

```diff
@@ -0,0 +1,223 @@
+package forth
+
+// Source: exercism/x-common
+// Commit: 8804a76 Remove non-ASCII characters tests from forth exercise
+
+var testCases = []testCase{
+	// parsing and numbers
+	{
+		"empty input results in empty stack",
+		[]string{},
+		[]int{},
+	},
+	{
+		"numbers just get pushed onto the stack",
+		[]string{"1 2 3 4 5"},
+		[]int{1, 2, 3, 4, 5},
+	},
+	{
+		"all non-word characters are separators",
+		[]string{"1\x002\x133\n4\r5 6\t7"},
+		[]int{1, 2, 3, 4, 5, 6, 7},
+	},
+
+	// addition
+	{
+		"can add two numbers",
+		[]string{"1 2 +"},
+		[]int{3},
+	},
+	{
+		"errors if there is nothing on the stack",
```

Comment:
> I thought it best to keep only one error in front of the student at one time.

Yeah, that seems like the way we like to do it here, so if possible we should try to maintain that property.

> Also, is there is a way to stop on first error when you have multiple TestXxxx functions?

Yeah, I hadn't found any I think, because `Fatalf` only stops the current test function, but still lets the other ones run.

> I'm thinking of two options: 1) Use the "sectioned" test cases again to allow the error output to indicate which section of the failed case

Since we've said we want to keep only one error for the student at a time, would these sections all be run from one function?

> 2) Update the canonical-data.json in x-common/exercises/forth to use more specific descriptions, like "+ causes error if nothing is on the stack", and "dup causes error if nothing is on the stack"

The interesting thing is having the descriptions like that make sense for test frameworks that automatically output the test output in a format like...

```
forth
  parsing and numbers
    empty input results in empty stack
    numbers just get pushed onto the stack
    all non-word characters are separators
  addition
    can add two numbers
    errors if there is nothing on the stack
    errors if there is only one value on the stack
  subtraction
    can subtract two numbers
    errors if there is nothing on the stack
    errors if there is only one value on the stack
```

I don't think we necessarily need to get that fancy, but what if, when we generate the test descriptions, we prepend the section to the description? I'll show in the code where I mean.